### PR TITLE
Move FastBoot and resource-discovery tests to node

### PR DESCRIPTION
**Pull request by chancancode**
Since these are server-side scenarios, we should be testing them in node
to more accurately replicated the target environment.

### CODE REVIEW

**Comment by chancancode** [comment]
cc @tomdale

**Comment by rwjblue** [comment]
I do agree that this should be tested in node, but it should also be tested in browser tests. 

* visit API is not a node only feature
* debugging is much easier in browser-land
* we ultimately intend to make ember-cli tests (that run in the browser) use the visit API in at least a few circumstances

**Comment by chancancode** [comment]
I kept the *unit* tests for visit in the browser side, and kept the browser-based integration scenario (Ember islands) in the browser suite. Do you think I should still keep a copy of the server integration scenarios in the browser suite? They were previously going out of their way to simulate the server environment (somewhat), so I removed them once I translated them into node as I don't see a huge benefit for them anymore.

I could be convinced to keep the duplicated tests based on the "easy to debug argument though.

**Comment by tomdale** [comment]
Surely there should be a way to create a suite of tests that get run in both node and the browser? Similar to how we can run the tests with and without `EXTEND_PROTOTYPES`, or how the metal tests are run both with a POJO and with an `Ember.Object`.

**Comment by rwjblue** [comment]
I am certain we could figure something out, but obviously changing the platform is a bit more difficult than the scenarios you listed.

**Comment by chancancode** [comment]
It isn't about creating the infrastructure to run the same code in both environment. The core issue here is that these **integration** tests are fundamentally testing the whole setup works correctly inside the specific environment they represent.

If you read the tests, you will see that they are not testing the `visit` feature – that's what the unit tests are for. These are higher level scenarios like server-side rendering and server-side resource-discovery. 

So while we could run them in the browser, it would requiring going out of the way to deliberately "mock out" certain features in unreliable ways.

If there are any holes that are not already unit-tested, we should probably fill the void with more unit test coverage. We should probably try to run more unit tests in both node and the browser where it makes sense. But both of those are pretty orthogonal to the move here.

The short version is that I basically took a short cut when I wrote these tests because I didn't have time to figure out how to test them in node; this pull request is basically to correct that mistake :P

**Comment by rwjblue** [comment]
@chancancode - I just reviewed the other visit unit tests (which I hadn't reviewed earlier) and this seems good to me.  A number of visit tests will continue to run in browser land ([see here](https://github.com/emberjs/ember.js/blob/0f5c7f5e750d3d1486ee2061fb10515c2f82282c/packages/ember-application/tests/system/visit_test.js#L83-L375)), only the "integration" tests are moving to node-land.

@tomdale - I'm :+1: on merging, are you :+1: or :-1: ?

**Comment by tomdale** [comment]
Big :+1:
